File: scripts/process_backend/_02/pb03/java/WriteFile.java
import java.io.*;
import java.nio.file.*;
import java.util.*;

public class WriteFile {
    public static void main(String[] args) throws Exception {
        int lines = args.length > 0 ? Integer.parseInt(args[0]) : 10000;
        String file = args.length > 1 ? args[1] : "output.csv";

        long start = System.currentTimeMillis();
        BufferedWriter writer = Files.newBufferedWriter(Paths.get(file));
        for (int i = 1; i <= lines; i++) {
            writer.write("Line " + i + ",Some data here\n");
        }
        writer.close();
        long end = System.currentTimeMillis();

        System.out.printf("[Java] Wrote %d lines to %s in %.4f seconds.%n", lines, file, (end - start) / 1000.0);
    }
}

File: scripts/process_backend/_02/pb03/cpp/write_file.cpp
#include <iostream>
#include <fstream>
#include <chrono>

int main(int argc, char* argv[]) {
    int lines = argc > 1 ? std::stoi(argv[1]) : 10000;
    std::string file = argc > 2 ? argv[2] : "output.csv";

    auto start = std::chrono::high_resolution_clock::now();
    std::ofstream ofs(file);
    for (int i = 1; i <= lines; ++i) {
        ofs << "Line " << i << ",Some data here\n";
    }
    ofs.close();
    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> duration = end - start;

    std::cout << "[C++] Wrote " << lines << " lines to " << file
              << " in " << duration.count() << " seconds." << std::endl;

    return 0;
}